TESTING.md

```diff
@@ -278,6 +280,129 @@ To run these tests, you go inside `test-artifacts` and execute:
 
 The execution then proceeds as explained in [Binary testing](#Binary-testing).
 
+## Building the test environment
+
+To run the tests, your vCD needs to have the following:
+
+* (1) An external network, with at least one, but preferably two allocatable IP addresses;
+* (2) An organization
+    * (3) An organization administrator user
+    * (4) A Virtual Data Center (VDC) with enough resources to build several vApps and VMs, and preferably 2 storage profiles
+        * (5) an Edge gateway with a default gateway
+    * (6) A catalog
+    * (7) A vApp template in the catalog
+
+Optionally, to run tests in [go-vcloud-director](https://github.com/vmware/go-vcloud-director), you will also need one or
+two Org Vdc networks and a media item.
+
+
+You can create all the above entities manually, or automate it using tools like Ansible or Puppet, or the [Vcloud Director
+command line utility](https://vmware.github.io/vcd-cli/). Starting with version 2.4.0 of this repository, however, you
+can also build the environment using Terraform itself. Here is how it works:
+
+The above entities are defined in the test configuration file (see `sample_vcd_test_config.json` for the full list of
+entities to provide). Here is an excerpt of where these names come from:
+
+```
+{
+  "vcd": {
+     "org": "datacloud",                    // (2)
+     "vdc": "datacloudvdc",                 // (4)
+     "catalog": {
+       "name": "datacloudcat",              // (6)
+       "catalogItem": "photon-hw11"         // (7)
+     },
+     [...]
+   },
+   "networking": {
+     "edgeGateway": "gwdatacloud",           // (5)
+     "externalNetwork": "extnetdatacloud",   // (1)
+     [...]
+     }
+   },
+   [...]
+ }
+
+```
+
+The vCD environment builder uses the test system information to create these entities. When triggered, it will take the
+names of the entities and some of the configuration parameters from the regular test configuration file. It will then
+integrate such data with additional information that is only used for the environment build, and it's listed under a new
+section in the file:
+
+```
+{
+   [...]
+  "testEnvBuild": {
+    // Indispensable data, which is not normally used in the tests but is mandatory to create the
+    // environment
+
+    "gateway": "GATEWAY_IP",
+    "netmask": "255.255.224.0",
+    "externalNetworkStartIp": "STARTING_IP",
+    "externalNetworkEndIp": "END_IP",
+    "dns1": "DNS_IP",
+    "mediaName": "test_media",
+
+
+    // Optional data. If not provided, the corresponding values
+    // from the rest of the configuration file will be used
+
+    "dns2": "8.8.4.4",
+    "externalNetworkPortGroup": "VM Network",
+    "externalNetworkPortGroupType": "NETWORK",
+    "ovaPath": "/path/to/photon-hw11-3.0-26156e2.ova",
+    "media_path":  "/path/to/test.iso",
+
+
+    // extra data. If not provided, these resources will not be created
```

Comment:
We should describe why and when would one want to pre-create these resources.

Reply:
done